### PR TITLE
New identities

### PR DESCRIPTION
**Pull request by Filmaluco**
This PR contains a few UI changes that were made along the way

### CODE REVIEW

File: app/src/main/java/org/stacks/app/domain/NewIdentity.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        val keys = generateNewIdentityKeys.generate()
+        val btcAddress = keys.toBtcAddress()
+        val identities = identityRepository.observe().first().toMutableList()
+
+        identities.add(generateIdentity.generate(btcAddress, value))
+
+        val profile = ProfileModel()
+        uploadProfile.upload(profile, keys)
+        uploadWallet.upload(identities)
+        identityRepository.set(identities)
```
Comment (Filmaluco) [inline app/src/main/java/org/stacks/app/domain/NewIdentity.kt:29]: Registar is missing! Will update

File: app/release/output-metadata.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (sdsantos) [inline app/release/output-metadata.json:1]: No need for this file to be on git.

File: app/src/main/java/org/stacks/app/domain/NewIdentity.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    private val registrarProfile: RegistrarProfile,
+    private val uploadProfile: UploadProfile,
+    private val uploadWallet: UploadWallet,
+) {
+    suspend fun create(username: String): Result<Unit> = try {
+        val keys = generateNewIdentityKeys.generate()
+        val btcAddress = keys.toBtcAddress()
+        val identities = identityRepository.observe().first().toMutableList()
+
+        identities.add(generateIdentity.generate(btcAddress, username))
```
Comment (sdsantos) [inline app/src/main/java/org/stacks/app/domain/NewIdentity.kt:23]: Not a bit deal, but whenever possible, avoid mutable versions. In this case:
```
val identities = identityRepository.observe().first() +
        generateIdentity.generate(btcAddress, username)
```

File: app/src/main/java/org/stacks/app/ui/auth/signup/ChooseUsernameViewModel.kt [hunk 4]
```diff
@@ -5,51 +5,77 @@ import kotlinx.coroutines.channels.BroadcastChannel
...
+        newAccount = signUp
+        usernameSubmitted.emit(username)
+    }
 
     // Outputs
     fun openNewAccountScreen() = openNewAccountScreen.asFlow()
+    fun loading() = loading.asStateFlow()
     fun errors() = errors.asFlow()
 
+    enum class Errors {
```
Comment (sdsantos) [inline app/src/main/java/org/stacks/app/ui/auth/signup/ChooseUsernameViewModel.kt:77]: Minor: should be just `Error`, like a super-class.

File: app/src/main/java/org/stacks/app/ui/auth/signup/ChooseUsernameViewModel.kt [hunk 5]
```diff
@@ -5,51 +5,77 @@ import kotlinx.coroutines.channels.BroadcastChannel
...
+    private val errors = BroadcastChannel<Errors>(1)
 
     init {
-        usernamePicked
-            .filter { it.isNotEmpty() }
-            .map { checkUsernameStatus.isAvailable(it) }
+        usernameSubmitted
+            .filter { it.isNotEmpty() && loading.value }
+            .map {
+                loading.emit(true)
```
Comment (sdsantos) [inline app/src/main/java/org/stacks/app/ui/auth/signup/ChooseUsernameViewModel.kt:37]: Didn't know about `emit()`, I've always used `.value =` 😛

File: app/src/main/java/org/stacks/app/ui/secret/bottomsheet/ShareKeyBottomSheetFragment.kt [hunk 6]
```diff
@@ -26,13 +26,18 @@ class ShareKeyBottomSheetFragment : BottomSheetDialogFragment() {
         ViewModelProvider(this).get(ShareKeyBottomSheetViewModel::class.java)
     }
 
+    var signUp: Boolean = false
```
Comment (sdsantos) [inline app/src/main/java/org/stacks/app/ui/secret/bottomsheet/ShareKeyBottomSheetFragment.kt:29]: I believe you can pass this through the Fragment constructor, and only create it on demand.

File: app/src/main/res/drawable/ic_rounded_bg.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="#FFFFFF"/>
+    <corners android:topRightRadius="10dp" android:topLeftRadius="10dp"/>
+    <padding android:left="0dp" android:top="0dp" android:right="0dp" android:bottom="0dp" />
```
Comment (sdsantos) [inline app/src/main/res/drawable/ic_rounded_bg.xml:5]: Se não tem padding, para que é preciso isto.

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1,14 +1,14 @@
 <resources>
     <!--APP-->
-    <string name="app_name">Connect</string>
+    <string name="app_name">Circle</string>
 
     <!--HOMEPAGE-->
-    <string name="connect_info">Connect guarantees your privacy by encrypting everything</string>
-    <string name="connect">Connect</string>
+    <string name="connect_info">Circle guarantees your privacy by encrypting everything</string>
+    <string name="circle">Circles</string>
```
Comment (sdsantos) [inline app/src/main/res/values/strings.xml:7]: Should be generic, not had the app name as key.

**Review by sdsantos — APPROVED** [review]

**Comment by Filmaluco** [comment]
Fixed with PR 25

**Pull request merged**